### PR TITLE
RPM probes - refactoring only

### PR DESCRIPTION
**Pull request by ybznek**
It is part of https://github.com/OpenSCAP/openscap/issues/379

Main reason to this pull request is - https://github.com/OpenSCAP/openscap/issues/368.
To add error handler to rpm functions I had to copy same function to all four probes - in rpm-helper.c file will be the callback function.

### CODE REVIEW

File: src/OVAL/probes/unix/linux/rpmverifypackage.c [hunk 1]
```diff
@@ -54,18 +54,7 @@
...
-#endif
-
-#ifndef HAVE_RPMFREECRYPTO
-# define rpmFreeCrypto() while(0)
-#endif
-
-#ifndef HAVE_RPMFREEFILESYSTEMS
-# define rpmFreeFilesystems() while(0)
-#endif
+#include "rpm-helper.h"
```
Comment (jan-cerny) [inline src/OVAL/probes/unix/linux/rpmverifypackage.c]: The removing should be part of the second commit.

Comment (ybznek) [inline src/OVAL/probes/unix/linux/rpmverifypackage.c]: Yes, sorry, oversight.

File: src/OVAL/probes/unix/linux/rpmverifypackage.c [hunk 2]
```diff
@@ -43,29 +43,16 @@
...
-
-#ifndef HAVE_RPMFREECRYPTO
-# define rpmFreeCrypto() while(0)
-#endif
-
-#ifndef HAVE_RPMFREEFILESYSTEMS
-# define rpmFreeFilesystems() while(0)
-#endif
+#define RPM_NOT_TOUCH
+#include "rpm-helper.h"
```
Comment (jan-cerny) [inline src/OVAL/probes/unix/linux/rpmverifypackage.c]: this include "rpm-helper.h" is twice here

File: src/OVAL/probes/unix/linux/rpmverifypackage.c [hunk 3]
```diff
@@ -43,29 +43,16 @@
...
-#endif
-
-#ifndef HAVE_RPMFREECRYPTO
-# define rpmFreeCrypto() while(0)
-#endif
-
-#ifndef HAVE_RPMFREEFILESYSTEMS
-# define rpmFreeFilesystems() while(0)
-#endif
+#define RPM_NOT_TOUCH
```
Comment (jan-cerny) [inline src/OVAL/probes/unix/linux/rpmverifypackage.c]: What is the meaning of RPM_NOT_TOUCH?

Comment (ybznek) [inline src/OVAL/probes/unix/linux/rpmverifypackage.c]: Original source code was disabling rpmFreeCrypto(),.. by macros. From some reason the disable code wasn't for this one function. I can investigate log of previous commits

Comment (ybznek) [inline src/OVAL/probes/unix/linux/rpmverifypackage.c]: I am wrong. I will fix it

**Comment by ybznek** [comment]
test this please

File: src/OVAL/probes/unix/linux/rpm-helper.c [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#include "rpm-helper.h"
```
Comment (jan-cerny) [inline src/OVAL/probes/unix/linux/rpm-helper.c]: Please add a license here.

**Comment by ybznek** [comment]
jenkins was offline
test this please

**Comment by jan-cerny** [comment]
@ybznek Thank you very much for the refactoring. Keep it coming! 

ACK.

**Pull request merged**